2263/1/about_scoring_project.rb

```diff
@@ -0,0 +1,121 @@
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+# rubocop:disable Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/AbcSize, Metrics/CyclomaticComplexity
+# Greed is a dice game where you roll up to five dice to accumulate
+# points.  The following "score" function will be used to calculate the
+# score of a single roll of the dice.
+#
+# A greed roll is scored as follows:
+#
+# * A set of three ones is 1000 points
+#
+# * A set of three numbers (other than ones) is worth 100 times the
+#   number. (e.g. three fives is 500 points).
+#
+# * A one (that is not part of a set of three) is worth 100 points.
+#
+# * A five (that is not part of a set of three) is worth 50 points.
+#
+# * Everything else is worth 0 points.
+#
+#
+# Examples:
+#
+# score([1,1,1,5,1]) => 1150 points
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+# :reek:TooManyStatements
+# :reek:UtilityFunction
+def score(dice)
+  score = 0
+  hash = { 1 => 0, 2 => 0, 3 => 0, 4 => 0, 5 => 0, 6 => 0 }
+  dice.each { |number| hash[number] += 1 }
+  hash.each do |key, number|
+    score += key * 100 + (number - 3) * 50 if key == 5 && number >= 3
```

Comment:
don't u want to move these number >= into separate if else?
Also, u can use `case number` to make this code more readable